src/components/button/Button.tsx

```diff
@@ -96,12 +101,27 @@ export function Button<T extends GroupTag = typeof DEFAULT_TAG>({
   spinnerColor,
   style,
   tag = DEFAULT_TAG,
+  testId,
   themeOverride,
   type = 'button',
   variant = 'standard',
   ...props
 }: ButtonProps<T>) {
   const Tag: GroupTag = tag;
+  const [buttonText, setButtonText] = useState('');
+
+  const handleButtonText = useCallback((node: Element | null) => {
+    setButtonText(node?.textContent || '');
+  }, []);
+
+  const defaultTestId = useMemo(
+    () =>
+      getTestId({
+        componentName: 'button',
+        name: buttonText,
+      }),
+    [buttonText],
+  );
```

Comment:
This useCallback hook approach doesn't seem to work, the callback would only trigger once, and when it's triggered, the ref in DOM may not be available to access yet.
<img width="304" alt="image" src="https://github.com/user-attachments/assets/07fbfa06-7ef8-4d66-a71b-55ac10dc22af">

Reply:
These are the right names. It's the text within the button appended to `amino--`. Unless you are referencing something else?